tensorflow/python/ops/image_ops_impl.py

```diff
@@ -3192,22 +3191,15 @@ def decode_image(contents,
     ValueError: On incorrect number of channels.
   """
   with ops.name_scope(name, 'decode_image'):
-    if compat.forward_compatible(2020, 8, 14):
-      channels = 0 if channels is None else channels
-      if dtype not in [dtypes.float32, dtypes.uint8, dtypes.uint16]:
-        dest_dtype = dtype
-        dtype = dtypes.uint16
-        return convert_image_dtype(gen_image_ops.decode_image(
-            contents=contents,
-            channels=channels,
-            expand_animations=expand_animations,
-            dtype=dtype), dest_dtype)
-      else:
-        return gen_image_ops.decode_image(
-            contents=contents,
-            channels=channels,
-            expand_animations=expand_animations,
-            dtype=dtype)
+    channels = 0 if channels is None else channels
+    if dtype not in [dtypes.float32, dtypes.uint8, dtypes.uint16]:
+      dest_dtype = dtype
+      dtype = dtypes.uint16
+      return convert_image_dtype(gen_image_ops.decode_image(
+          contents=contents,
+          channels=channels,
+          expand_animations=expand_animations,
+          dtype=dtype), dest_dtype)
 
```

Comment:
You would need to remove lines 3204-3276 and remove line 3195, and keep the rest (with proper indentation). The else block removed above is not the else case for `if compat.forward_compatible(2020, 8, 14)`.

Reply:
Good catch, sorry about that. Looks like my editor got confused with the 2-space indent.

Reply:
Thank you for the quick fix!